sale_order_report_delivered/models/sale_order.py

```diff
@@ -0,0 +1,54 @@
+# © 2023 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+from functools import partial
+from odoo.tools.misc import formatLang
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def get_qty_delivered_lines(self):
+        line_ids = []
+        for line in self.order_line:
+            sol_vals = line._convert_to_write(line._cache)
+            valued_line = line.new(sol_vals)
+            valued_line.product_uom_qty = line.qty_delivered
+            valued_line.tax_id = line.tax_id
+            line_ids.append(valued_line)
+        return line_ids
+
+    def get_qty_delivered_subtotal(self):
+        price_subtotal = 0
+        for line in self.get_qty_delivered_lines():
+            price_subtotal += line.price_subtotal
+        return price_subtotal
+
+    def get_qty_delivered_taxes(self):
```

Comment:
This code, is inspired/copied from which Odoo file? 😇
I would take a look a it

Reply:
Its a copy of this: https://github.com/odoo/odoo/blob/ab0ae8dadf55baa258d733835acc3a12771932be/addons/sale/models/sale.py#L983
But with the temp lines